tools/unit-tests-merge.cs

```diff
@@ -0,0 +1,105 @@
+    if (files.Length == 0) {
+        Console.WriteLine ($"Not files were found with the pattern '{testFilePattern}'");
+        Environment.Exit (-1);
+    }
+
+    if (files.Length == 1) {
+        // we only have a single tests, so we do not need to do anything
+        Console.WriteLine ("Only one file was found, doing nothing.");
+        Environment.Exit (0);
+    }
+
+    // get the total errors, failures, etc.. from all the files, this will be used to later
+    // write the main test-results node using the addition of all of them
+    var errors = 0L;
+    var inconclusive = 0L;
+    var ignored = 0L;
+    var invalid = 0L;
+    var notRun = 0L;
+    var total = 0L;
+    var failures = 0L;
+    var skipped = 0L;
+    var date = DateTime.Today;
+    TimeSpan? time = null;
+
+    foreach ( var filePath in files) {
+        using (var stream = new StreamReader (filePath))
+        using (var reader = XmlReader.Create (stream)) {
+            while (reader.Read ()) {
+                if (reader.NodeType == XmlNodeType.Element && reader.Name == "test-results") {
+                    // read the attrs and get out of the loop
+                    errors += Convert.ToInt64 (reader ["errors"]);
+                    inconclusive += Convert.ToInt64 (reader ["inconclusive"]);
+                    ignored += Convert.ToInt64 (reader ["ignored"]);
+                    invalid += Convert.ToInt64 (reader ["invalid"]);
+                    notRun += Convert.ToInt64 (reader ["not-run"]);
+                    total += Convert.ToInt64 (reader ["total"]);
+                    failures += Convert.ToInt64 (reader ["failures"]);
+                    skipped += Convert.ToInt64 (reader ["skipped"]);
+                    if (time.HasValue)
+                        time = time.Value.Add (TimeSpan.Parse (reader ["time"]));
+                    else
+                        time = TimeSpan.Parse (reader ["time"]);
+                    break; // while loop exit
+                }
+            }
+        }
+    }
+
+    // create a writer for the final solution
+    var outPutPath = "tests-grouped-result.xml";
+
+    if (File.Exists (outPutPath))
+        File.Delete (outPutPath);
+
+    using (var writer = File.CreateText (outPutPath)) {
+        writer.WriteLine ("<?xml version=\"1.0\" encoding=\"utf-8\"?>");
+        writer.WriteLine ($"<test-results name=\"Test results\" errors=\"{errors}\" inconclusive=\"{inconclusive}\" ignored=\"{ignored}\" invalid=\"{invalid}\" not-run=\"{notRun}\" date=\"{date}\" time=\"{time.Value}\" total=\"{total}\" failures=\"{failures}\" skipped=\"{skipped}\">");
+        writer.WriteLine ("<environment os-version=\"unknown\" platform=\"unknown\" cwd=\"unknown\" machine-name=\"unknown\" user=\"unknown\" user-domain=\"unknown\" nunit-version=\"xUnit.net 2.4.0.4049\" clr-version=\"64-bit .NET Standard [collection-per-class, non-parallel]\"></environment>");
```

Comment:
Can you get all the `unknown`s from any of the input xmls?

Reply:
@rolfbjarne that is something I did not like, and unfortunately, I cannot. But I'm going to fix the runner to report that. I want to see the environment and include the touch args used. We will need to do another PR (we first need to see if VSTS is happy with the HUGE file is going to generate.